### PR TITLE
ROX-10298: Support new policy categories on Policy Wizard step 1

### PR DESCRIPTION
**Pull request by pedrottimark**
## Description

1. Add `'ROX_POSTGRES_DATASTORE'` in addition to `'ROX_NEW_POLICY_CATEGORIES'` feature flag.

2. Conditionally fetch policy categories.
    * If enabled: call `getPolicyCategories` function from PolicyCategoriesService service.
    * If disabled: call renamed `getPolicyCategoriesNonPostgres` function from Policies service and use `name` as `id` property to keep equivalent `key` prop for select items.

3. Prevent creation of new categories if enabled:
    * `onCreateOption` use default value from PatternFly doc page
        https://www.patternfly.org/v4/components/select#select
    * `isCreatable` is false

### Residue

1. Investigate why duplicate 3 requests before and after /v1/policies/id request in step 2 below

2. Visit /main/policy-management/policy-categories without feature flag
    * See empty page instead of 404 page

3. Visit /main/policy-management/policies
    * See **Policy management - Policy categories** in title on browser tab with or without feature flag

## Checklist
- [ ] Investigated and inspected CI test results
- [x] ~Unit test and regression tests added~

### With ROX_NEW_POLICY_CATEGORIES and ROX_POSTGRES_DATASTORE

1. Visit /main/policy-management/policy-categories
    * See /v1/policycategories request
        1_Policy_categories

2. Visit Visit /main/policy-management/policies and then edit a policy
    * See /v1/policycategories request **Residue** question above
        2_Edit_policy

3. Create a new policy
    * See categories
        3_Cteate_Policy

4. View created policy
    * See categories
        4_View_created

5. Edit created policy
    * Remove categories, and then select different category.
    * Type a non-existent category, and then click Enter. Nothing happens.
        5_Select_categories

6. Save edited policy
    * See category
        6_Save_edited

### Without ROX_NEW_POLICY_CATEGORIES

1. Visit /main/policy-management
    * See no **Policy categories** tab
        1_no_categories

2. Visit /main/policy-management/policies and then edit a policy
    * See /v1/policyCategories request **Residue** question above
        2_policyCategories

3. Create a new policy
    * See categories
        3_Save_created

4. View created policy
    * See categories
        4_View

5. Edit created policy
    * Remove categories, and then select different category.
    * Type a non-existent category, and then see **Create whatever**. Click to create it.
        5_isCreatable

6. Save edited policy
    * See category in PUT /v1/policies/id payload
        6_Save_new_category

7. Click **Edit** again
    * See new category in GET /v1/policyCategories response
        7_policyCategories

### CODE REVIEW

File: ui/apps/platform/src/Containers/Policies/Wizard/Step1/PolicyCategoriesSelectField.tsx [hunk 1]
```diff
@@ -60,12 +76,14 @@ function PolicyCategoriesSelectField(): ReactElement {
                 selections={field.value}
                 onSelect={onSelectHandler(field.value)}
                 onToggle={onCategoriesToggle}
-                onCreateOption={onCreateCategory}
+                onCreateOption={
+                    isPolicyCategoriesEnabled ? () => undefined as void : onCreateCategory
+                }
```
Comment (alwayshooin) [inline ui/apps/platform/src/Containers/Policies/Wizard/Step1/PolicyCategoriesSelectField.tsx:81]: oh just curious why the `as void` is necessary here. can you not just do `() => {}` ?

File: ui/apps/platform/src/Containers/Policies/Wizard/Step1/PolicyCategoriesSelectField.tsx [hunk 2]
```diff
@@ -60,12 +76,14 @@ function PolicyCategoriesSelectField(): ReactElement {
...
                 onClear={clearSelection}
-                isCreatable
+                isCreatable={!isPolicyCategoriesEnabled}
             >
-                {policyCategories.map((category) => (
-                    <SelectOption key={category} value={category} />
+                {policyCategories.map(({ id, name }) => (
+                    <SelectOption key={id} value={name} />
```
Comment (alwayshooin) [inline ui/apps/platform/src/Containers/Policies/Wizard/Step1/PolicyCategoriesSelectField.tsx:86]: I think you might have to conditionally do the `key={category} value={category}` mapping here based off of the feature flag

File: ui/apps/platform/src/Containers/Policies/Wizard/Step1/PolicyCategoriesSelectField.tsx (hunk 1 above)
Comment (pedrottimark) [reply ui/apps/platform/src/Containers/Policies/Wizard/Step1/PolicyCategoriesSelectField.tsx:81]: Good question. Copied from PatternFly docs.

**Review by alwayshooin — APPROVED** [review]
code looks good to me! 🚢

**Pull request merged**